packages/ui-components/src/components/Switch/Switch.component.tsx

```diff
@@ -0,0 +1,310 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, useMemo, useId } from "react"
+
+import { Label } from "../Label/Label.component"
+import { Icon } from "../Icon/Icon.component"
+import { FormHint } from "../FormHint/FormHint.component"
+
+const switchBaseStyles = `
+    jn-rounded-full
+    jn-relative
+    jn-p-0
+    jn-leading-0
+    jn-border
+    jn-g-theme-default
+    focus:jn-outline-none
+    focus:jn-ring-2
+    focus:jn-ring-theme-focus
+    disabled:jn-opacity-50
+    disabled:jn-cursor-not-allowed
+`
+
+const borderBaseStyles = `
+    jn-border-theme-switch-default
+`
+
+const switchWrapperStyles = `
+    jn-flex
+    jn-flex-row
+    jn-items-center
+`
+
+const handleBaseStyles = `
+    jn-inline-block
+    jn-absolute
+    jn-top-[1px]
+    jn-rounded-full
+    jn-bg-theme-switch-handle
+    jn-border-theme-default
+`
+
+const handleOnStyles = `
+    jn-right-[1px] 
+    jn-bg-theme-switch-handle-checked
+`
+
+const handleOffStyles = `
+    jn-left-[1px]
+`
+
+const validBaseStyles = `
+    jn-border-theme-success
+`
+
+const invalidBaseStyles = `
+    jn-border-theme-error
+`
+
+const iconBaseStyles = `
+    jn-inline-block 
+    jn-ml-1 
+    jn-leading-1
+    jn-mt-[-.2rem]
+`
+
+const hintBaseStyles = `
+    jn-mt-0
+`
+
+type SwitchSize = "small" | "default" | "large"
+
+export interface SwitchProps extends Omit<React.ComponentPropsWithoutRef<"button">, "ref"> {
+  /**
+   * HTML name attribute for the switch button
+   */
+  name?: string
+  /**
+   * HTML id attribute for the switch button
+   */
+  id?: string
+  /**
+   * Label to display next to the Switch component
+   */
+  label?: string
+  /**
+   * Indicates if the Switch is required
+   */
+  required?: boolean
+  /**
+   * Specifies the size of the Switch component
+   */
+  size?: SwitchSize
+  /**
+   * Sets the initial checked state of the Switch component for use in uncontrolled mode
+   */
+  on?: boolean
+  /**
+   * Disabled state of the Switch component
+   */
+  disabled?: boolean
+  /**
+   * Indicates if the Switch has an error state
+   */
+  invalid?: boolean
+  /**
+   * Indicates if the Switch has been validated successfully
+   */
+  valid?: boolean
+  /**
+   * Help text to provide additional information about the Switch
+   */
+  helptext?: React.ReactNode
+  /**
+   * Text to display when the Switch has validation errors
+   */
+  errortext?: React.ReactNode
+  /**
+   * Text to display when the Switch has been validated successfully
+   */
+  successtext?: React.ReactNode
+  /**
+   * Custom class name to apply to the internal button element
+   */
+  className?: string
+  /**
+   * Callback function to handle changes to the switch state
+   */
+  onChange?: (_event: React.ChangeEvent<HTMLButtonElement>) => void
```

Comment:
I would really reconsider the _ and use eslint ignore instead of poluting types with _ cause it is really old practice, I think we need to find a proper solution for that and meanwhile just use the eslintignore

Reply:
let's remove the rule then https://github.com/cloudoperators/juno/blob/main/packages/config/eslint/juno-typescript.mjs#L60-L72

Reply:
It is a tricky, I would like to keep a rule but tweak it for TS, so no rule on types, we need to investigate on it, since my last attemtion there were no possibility

Reply:
Deal, will add ignore comment. Thank You!